File: leetcode/algorithm/countBits.c
#include <stdio.h>
#include <stdlib.h>

int getBits(int n){
	int ret = 0;
	while (n != 0){
		n = n & (n-1);
		ret++;
	}
	return ret;
}

int* countBits(int num, int* returnSize) {
	int *ret = (int*)malloc((num+1) * sizeof(int));
	*returnSize = num+1;
    for(int i = 0; i <= num; i++){
		ret[i] = getBits(i);
	}
	
	return ret;
}

int main(){
	int *ret, nums;
	ret = countBits(5, &nums);
	printf("%d\n", nums);
	for (int i = 0; i < nums; i++){
		printf("%d ", ret[i]);
	}
	printf("\n");
}

File: leetcode/algorithm/countBits.go
package main

import (
	"fmt"
)

func getBits(n int) int {
	ret := 0
	for n != 0 {
		n = n & (n - 1)
		ret++
	}
	return ret
}

func countBits(num int) []int {
	var ret []int
	for i := 0; i <= num; i++ {
		ret = append(ret, getBits(i))
	}
	return ret
}

func main() {
	ret := countBits(5)
	fmt.Println(ret)

	ret = countBits(123456)
	fmt.Println("finished, length of ret is", len(ret))
}
